sphinx/texinputs/sphinx.sty

```diff
@@ -682,13 +710,28 @@
 % for captions of literal blocks
 % also define `\theH...` macros for hyperref
 \newcounter{literalblock}
+\ifspx@opt@nonumfigreset
+  \def\theliteralblock {\arabic{literalblock}}
+  \def\theHliteralblock{\arabic{literalblock}}
+\else
+\ifx\spx@opt@numfigreset\@empty
 \ltx@ifundefined{c@chapter}
   {\@addtoreset{literalblock}{section}
   \def\theliteralblock {\ifnum\c@section>\z@ \thesection.\fi\arabic{literalblock}}
   \def\theHliteralblock {\theHsection.\arabic{literalblock}}}
   {\@addtoreset{literalblock}{chapter}
   \def\theliteralblock {\ifnum\c@chapter>\z@ \thechapter.\fi\arabic{literalblock}}
   \def\theHliteralblock {\theHchapter.\arabic{literalblock}}}
+\else
+  \@addtoreset{literalblock}{\spx@opt@numfigreset}
+  \edef\theliteralblock
+      {\noexpand\ifnum\@nameuse{c@\spx@opt@numfigreset}>\z@
+       \expandafter\noexpand\csname the\spx@opt@numfigreset\endcsname
+       .\noexpand\fi\noexpand\arabic{literalblock}}
+  \edef\theHliteralblock
+      {\expandafter\noexpand\csname theH\spx@opt@numfigreset\endcsname
+       .\noexpand\arabic{literalblock}}
+\fi\fi
```

Comment:
perhaps this avoidance of printing ``0.`` should be applied only at start of document. If for example the numfig reset happens at subsections, this means if in a section a literalblock arises before first subsection, then it drops entirely the prefix numbering, which might be confusing, the number will be typically ``1``, and then the first one in coming subsection will be ``1.1.1``. Perhaps better for former one to be numbered ``1.0.1``

Reply:
I was actually referring to code about 5 lines above that location, sorry.